### ISSUE TITLE
Android: ANR on Release with minSdkVersion=23 (Android 6)

### ISSUE THREAD

**Issue by JFMG** [CrossGeeks/FirebasePushNotificationPlugin#179]
## Bug Information
Version Number of Plugin: 2.3.5
Device Tested On: Samsung Galaxy On5 (Android 6.0), LG Nexus 5X (Android 8.1)
Simulator Tested On: none
Version of VS: Visual Studio for Mac (stable channel)
Version of Xamarin: stable channel
Version of other libs: Xamarin.Forms 3.0

### Steps to reproduce the Behavior
In MainApplication.cs I override OnCreate and call `FirebasePushNotificationManager.Initialize(this, false);` I set the configuration to Release and deploy on a device. The minSdkVersion in the AndroidManifest.xml is set to 23. I found out it works fine with minSdkVersion 22 or below. Attention: You will need to delete bin and obj folders for a correct reproduction.

### Expected Behavior
The app starts and does not freeze.

### Actual Behavior
I get an ANR (application not responding) error. The app freezes while showing the splash screen

### Logcat
https://gist.github.com/JFMG/259612a412712eae8e51bd792769b1ba

**Comment by JFMG** [comment]
## Update

I got the following from enabling the Debug output:

```
[FirebaseInstanceId] background sync failed: TIMEOUT, retry in 10s

PushNotificationService.OnNotificationError Error type: UnregistrationFailed Error: Java.IO.IOException: TIMEOUT
  at Java.Interop.JniEnvironment+InstanceMethods.CallVoidMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniMethodInfo method) [0x00068] in <8379a5aaffdb444fba160f472bfa8844>:0 
  at Android.Runtime.JNIEnv.CallVoidMethod (System.IntPtr jobject, System.IntPtr jmethod) [0x0000e] in <68250d09e80546e08276514979c53f5e>:0 
  at Firebase.Iid.FirebaseInstanceId.DeleteInstanceId () [0x00043] in <c0531c685a2b4e92a77d8ce91331354b>:0 
  at Plugin.FirebasePushNotification.FirebasePushNotificationManager.CleanUp (System.Boolean clearAll) [0x00012] in <fb36c9519f0f4c3b868ba010c95b7485>:0 
  at Plugin.FirebasePushNotification.FirebasePushNotificationManager+<>c__DisplayClass55_0.<Initialize>b__0 (System.Object state) [0x000f4] in <fb36c9519f0f4c3b868ba010c95b7485>:0 
  --- End of managed Java.IO.IOException stack trace ---
java.io.IOException: TIMEOUT
	at com.google.firebase.iid.zzo.zzcgd(Unknown Source:21)
	at com.google.firebase.iid.zzl.zzb(Unknown Source:456)
	at com.google.firebase.iid.zzl.zza(Unknown Source:14)
	at com.google.firebase.iid.zzj.zzb(Unknown Source:38)
	at com.google.firebase.iid.zzj.zza(Unknown Source:39)
	at com.google.firebase.iid.FirebaseInstanceId.deleteInstanceId(Unknown Source:7)
```

**Comment by JFMG** [comment]
I found the bug. It is because I don't request permissions and has nothing to do with this plugin. I am sorry for the trouble. I will close this issue.

For everyone who is interested in my problem, it is described here: https://github.com/xamarin/xamarin-android/issues/1163

Summary: When you set the Target SDK version to latest in your Android Manifest file, it will get removed. But this means that the Target versions is set to the minimum version. After setting the minimum version to 23 (Android 6), I had to use permission requests. I do it now and my app works again.
